packages/valory/skills/mech_interact_abci/behaviours.py

```diff
@@ -0,0 +1,749 @@
+        self._from_block = from_block
+
+    @property
+    def request_id(self) -> int:
+        """Get the request id."""
+        return self._request_id
+
+    @request_id.setter
+    def request_id(self, request_id: Union[str, int]) -> None:
+        """Set the request id."""
+        try:
+            self._request_id = int(request_id)
+        except ValueError:
+            msg = f"Request id {request_id} is not a valid integer!"
+            self.context.logger.error(msg)
+
+    @property
+    def response_hex(self) -> str:
+        """Get the hash of the response data."""
+        return self._response_hex
+
+    @response_hex.setter
+    def response_hex(self, response_hash: bytes) -> None:
+        """Set the hash of the response data."""
+        try:
+            self._response_hex = response_hash.hex()
+        except AttributeError:
+            msg = f"Response hash {response_hash!r} is not valid hex bytes!"
+            self.context.logger.error(msg)
+
+    @property
+    def mech_response_api(self) -> MechResponseSpecs:
+        """Get the mech response api specs."""
+        return self.context.mech_response
+
+    def set_mech_response_specs(self) -> None:
+        """Set the mech's response specs."""
+        full_ipfs_hash = IPFS_HASH_PREFIX + self.response_hex
+        ipfs_link = self.params.ipfs_address + full_ipfs_hash + f"/{self.request_id}"
+        # The url must be dynamically generated as it depends on the ipfs hash
+        self.mech_response_api.__dict__["_frozen"] = False
+        self.mech_response_api.url = ipfs_link
+        self.mech_response_api.__dict__["_frozen"] = True
+
+    @property
+    def mech_response(self) -> MechInteractionResponse:
+        """Get the mech response api specs."""
+        if self._mech_response is None:
+            error = "The mech's response has not been set!"
+            return MechInteractionResponse(error=error)
+        return self._mech_response
+
+    def _get_block_number(self) -> WaitableConditionType:
+        """Get the block number in which the request to the mech was settled."""
+        result = yield from self.contract_interact(
+            performative=ContractApiMessage.Performative.GET_RAW_TRANSACTION,  # type: ignore
+            # we do not need the address to get the block number, but the base method does
+            contract_address=ZERO_ADDRESS,
+            contract_public_id=Mech.contract_id,
+            contract_callable="get_block_number",
+            data_key="number",
+            placeholder=get_name(MechResponseBehaviour.from_block),
+            tx_hash=self.synchronized_data.final_tx_hash,
+        )
+
+        return result
+
+    def _get_request_id(self) -> WaitableConditionType:
+        """Get the request id."""
+        result = yield from self._mech_contract_interact(
+            contract_callable="process_request_event",
+            data_key="requestId",
+            placeholder=get_name(MechResponseBehaviour.request_id),
```

Comment:
The multisend is not for a single request. However, the `MechResponseBehaviour`'s implementation makes this assumption. For example, here, the `request_id` should not be one.